### DISCUSSION TITLE
Disable auto sign in if sign up successful

### DISCUSSION THREAD

**Discussion by A-L-M** [supabase/supabase#11909] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I'm building an admin panel that lets a signed-in user create new users with an email and password. The problem is that calling the `supabase.auth.signUp(...)` function changes the current user to the newly created one automatically. Before, I tried using `supabase.auth.api.signUpWithEmail(...)` according to [this discussion](https://github.com/supabase/supabase/discussions/5784). This function is no longer available as far as I know.

**Describe the solution you'd like**
I'd like an alternative to the `supabase.auth.api.signUpWithEmail(...)` function that existed before.

**Comment by dshukertjr** [comment]
Transferring this issue to discussions to further discuss this feature.

Currently, you have the following workaround for it. 
- You can directly call the signUp endpoint. You can see the endpoint and parameters [here](https://github.com/supabase/gotrue-dart/blob/main/lib/src/gotrue_client.dart#L120).
- You can use the [admin API](https://github.com/supabase/gotrue-dart/blob/main/lib/src/gotrue_admin_api.dart#L46) for creating users from a secure server environment.

**Comment by guizoxxv** [comment]
@A-L-M,
You may be able to solve your case using the [signInWithOtp](https://supabase.com/docs/reference/javascript/auth-signinwithotp) function. 

I have a similar issue, but it cannot be solved with this function. In my app, an admin is creating users using the `signUp` function. The thing is that I don't want to send the magic link when the admin creates the user, only when the user tries to login.
